app/src/main/java/is/xyz/mpv/MPVActivity.kt

```diff
@@ -774,6 +774,12 @@ class MPVActivity : AppCompatActivity(), MPVLib.EventObserver, TouchGesturesObse
         if (lockedUI)
             return showUnlockControls()
 
+        val prefs = getDefaultSharedPreferences(applicationContext)
+        if (prefs.getBoolean("save_position", false)) {
+            finishWithResult(RESULT_OK, true)
+            return
+        }
```

Comment:
you can directly retrieve the settings from `this.shouldSavePosition`
and also it'd be better to integrate it with if-clause that already exists

Reply:
Thank you for the comment!
Is there any other reason to give the exit warning? Maybe better off to have a user config option to disable it.

Reply:
The warning is so you don't lose your playlist, I don't think there's a reason to disable it right now.

Reply:
That's my second thought after the PR. Then this PR is irrelevant as my use case is directory playlist.